PY3_DLLNAME not correct?

I noticed a warning in the ARM64 build:
```
In file included from <built-in>:374:
<command line>:5:9: warning: 'PY3_DLLNAME' macro redefined [-Wmacro-redefined]
#define PY3_DLLNAME L"libpython3.10.dll"
        ^
<command line>:3:9: note: previous definition is here
#define PY3_DLLNAME "libpython3.dll"
        ^
```

There was another warning elsewhere:
```
../Python/pathconfig.c:338:45: warning: incompatible pointer types passing 'char [15]' to parameter of type 'LPCWSTR' (aka 'const unsigned short *') [-Wincompatible-pointer-types]
        HMODULE hPython3 = GetModuleHandleW(PY3_DLLNAME);
                                            ^~~~~~~~~~~
<command line>:3:21: note: expanded from here
#define PY3_DLLNAME "libpython3.dll"
                    ^~~~~~~~~~~~~~~~
C:/msys64/clangarm64/aarch64-w64-mingw32/include/libloaderapi.h:140:53: note: passing argument to parameter 'lpModuleName' here
WINBASEAPI HMODULE WINAPI GetModuleHandleW (LPCWSTR lpModuleName);
```

I think that `PY3_DLLNAME` should be `L"libpython3.10.dll"` not `"libpython3.dll"`

--- a/configure.ac
+++ b/configure.ac
@@ -3961,7 +3961,7 @@ then
 	*-*-mingw*)
 	DYNLOADFILE="dynload_win.o"
 	extra_machdep_objs="$extra_machdep_objs PC/dl_nt.o"
-	CFLAGS_NODIST="$CFLAGS_NODIST -DMS_DLL_ID='\"$VERSION\"' -DPY3_DLLNAME='\"libpython3.dll\"'"
+	CFLAGS_NODIST="$CFLAGS_NODIST -DMS_DLL_ID='\"$VERSION\"' -DPY3_DLLNAME='L\"$DLLLIBRARY\"'"
 	;;
 	esac
 fi

